<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CLIP-Seq Workflow Template • systemPipeCLIPseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="CLIP-Seq Workflow Template">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158796879-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158796879-1');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeCLIPseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a></a>
</li>
<li>
  <a href="../articles/systemPipeCLIPseq.html">Get started</a>
</li>
<li>
  <a href="../articles/systemPipeCLIPseq.html">
    <span class="fas fa fas fa-book"></span>
     
    Workflow
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tgirke/systemPipeR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>CLIP-Seq Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 22 February, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/systemPipeCLIPseq/blob/master/vignettes/systemPipeCLIPseq.Rmd"><code>vignettes/systemPipeCLIPseq.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeCLIPseq.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!-- 
Compile from command-line  
Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeChIPseq.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeChIPseq.knit.md 13; Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::pdf_document'))"
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Users want to provide here background information about the CLIP-Seq project.</p>
<div id="background-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#background-and-objectives" class="anchor"></a>Background and objectives</h2>
<p>This report describes the analysis of several e-CLIP experiments studying the RNA biding proteins.</p>
</div>
<div id="experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-design" class="anchor"></a>Experimental design</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div id="workflow-environment" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-environment" class="anchor"></a>Workflow environment</h1>
<div id="generate-workflow-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-workflow-environment" class="anchor"></a>Generate workflow environment</h2>
</div>
<div id="run-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#run-workflow" class="anchor"></a>Run workflow</h2>
<div id="run-r-session-on-computer-node" class="section level3">
<h3 class="hasAnchor">
<a href="#run-r-session-on-computer-node" class="anchor"></a>Run R session on computer node</h3>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeR)</a></code></pre></div>
</div>
</div>
</div>
<div id="read-preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#read-preprocessing" class="anchor"></a>Read preprocessing</h1>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">targetspath &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, <span class="dt">package=</span><span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment.char =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">targets</a></code></pre></div>
<pre><code>##                       FileName SampleName Factor SampleLong Experiment        Date
## 1  ./data/SRR446027_1.fastq.gz        M1A     M1  Mock.1h.A          1 23-Mar-2012
## 2  ./data/SRR446028_1.fastq.gz        M1B     M1  Mock.1h.B          1 23-Mar-2012
## 3  ./data/SRR446029_1.fastq.gz        A1A     A1   Avr.1h.A          1 23-Mar-2012
## 4  ./data/SRR446030_1.fastq.gz        A1B     A1   Avr.1h.B          1 23-Mar-2012
## 5  ./data/SRR446031_1.fastq.gz        V1A     V1   Vir.1h.A          1 23-Mar-2012
## 6  ./data/SRR446032_1.fastq.gz        V1B     V1   Vir.1h.B          1 23-Mar-2012
## 7  ./data/SRR446033_1.fastq.gz        M6A     M6  Mock.6h.A          1 23-Mar-2012
## 8  ./data/SRR446034_1.fastq.gz        M6B     M6  Mock.6h.B          1 23-Mar-2012
## 9  ./data/SRR446035_1.fastq.gz        A6A     A6   Avr.6h.A          1 23-Mar-2012
## 10 ./data/SRR446036_1.fastq.gz        A6B     A6   Avr.6h.B          1 23-Mar-2012
## 11 ./data/SRR446037_1.fastq.gz        V6A     V6   Vir.6h.A          1 23-Mar-2012
## 12 ./data/SRR446038_1.fastq.gz        V6B     V6   Vir.6h.B          1 23-Mar-2012
## 13 ./data/SRR446039_1.fastq.gz       M12A    M12 Mock.12h.A          1 23-Mar-2012
## 14 ./data/SRR446040_1.fastq.gz       M12B    M12 Mock.12h.B          1 23-Mar-2012
## 15 ./data/SRR446041_1.fastq.gz       A12A    A12  Avr.12h.A          1 23-Mar-2012
## 16 ./data/SRR446042_1.fastq.gz       A12B    A12  Avr.12h.B          1 23-Mar-2012
## 17 ./data/SRR446043_1.fastq.gz       V12A    V12  Vir.12h.A          1 23-Mar-2012
## 18 ./data/SRR446044_1.fastq.gz       V12B    V12  Vir.12h.B          1 23-Mar-2012</code></pre>
</div>
<div id="fastq-quality-report" class="section level2">
<h2 class="hasAnchor">
<a href="#fastq-quality-report" class="anchor"></a>FASTQ quality report</h2>
<p>The following <code>seeFastq</code> and <code>seeFastqPlot</code> functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box plots, base proportions, base-level quality trends, relative k-mer diversity, length and occurrence distribution of reads, number of reads above quality cutoffs and mean quality distribution. The results are written to a PDF file named <code>fastqReport.pdf</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/trim_clipper.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets.txt"</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">fqlist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastq</a></span>(<span class="dt">fastq=</span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(args), <span class="dt">batchsize=</span><span class="dv">100000</span>, <span class="dt">klength=</span><span class="dv">8</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/fastqReport.pdf"</span>, <span class="dt">height=</span><span class="dv">18</span>, <span class="dt">width=</span><span class="dv">4</span><span class="op">*</span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(fqlist))</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastqPlot</a></span>(fqlist)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/fastqReport.png"><div align="center">
Figure 1: FASTQ quality report for 2 samples
</div>
<p></p>
</div>
<div id="read-quality-filtering-and-trimming" class="section level2">
<h2 class="hasAnchor">
<a href="#read-quality-filtering-and-trimming" class="anchor"></a>Read quality filtering and trimming</h2>
<div id="remove-adaptor-with-fastx_clipper" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-adaptor-with-fastx_clipper" class="anchor"></a>Remove Adaptor with fastx_clipper</h3>
<p>Removing Adapter sequence using fastx_toolkit. OBS: The adapter at the 3’ end should be AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC and the 10nt on the 5’ end of the read are the barcode.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/trim_clipper.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets.txt"</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x=</span>args, <span class="dt">file=</span><span class="st">"targets_clip.txt"</span>, <span class="dt">overwrite=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<div id="first-look-at-the-data" class="section level4">
<h4 class="hasAnchor">
<a href="#first-look-at-the-data" class="anchor"></a>First look at the data</h4>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"seeDistribution.R"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"results/Distribution_AdapterRemoval.pdf"</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">seeDistribution</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span>(args), <span class="dt">file=</span><span class="st">"fastq"</span>, <span class="dt">batchsize =</span> <span class="dv">100000</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
</div>
</div>
<div id="mapping-all-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-all-reads" class="anchor"></a>Mapping ALL reads</h2>
<p>In order to have a better understanding of the data, it’s possible to map all the reads with the reference genome.</p>
<div id="remove-barcode" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-barcode" class="anchor"></a>Remove barcode</h3>
<p>Second step, remove barcode. In this data, N = 10.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/trim.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets_clip.txt"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args)</a></code></pre></div>
<div id="reads-length-distribution" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after barcode removal.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"seeDistribution.R"</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"results/Distribution_barcodeRemoval.pdf"</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">seeDistribution</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span>(args), <span class="dt">file=</span><span class="st">"fastq"</span>, <span class="dt">batchsize =</span> <span class="dv">100000</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
</div>
<div id="read-mapping-with-hisat2" class="section level3">
<h3 class="hasAnchor">
<a href="#read-mapping-with-hisat2" class="anchor"></a>Read mapping with <code>HISAT2</code>
</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/hisat2.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets_clip_trim.txt"</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span>(args)[<span class="dv">1</span>] <span class="co"># Command-line parameters for first FASTQ file</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span>(args))</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">resources &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">walltime=</span><span class="dv">1440</span>, <span class="dt">ntasks=</span><span class="dv">1</span>, <span class="dt">ncpus=</span><span class="dv">12</span>, <span class="dt">memory=</span><span class="dv">10240</span>, <span class="dt">email=</span><span class="st">"danielac@ucr.edu"</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">hisat2 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/clusterRun.html">clusterRun</a></span>(args, <span class="dt">conffile=</span><span class="st">".batchtools.conf.R"</span>, <span class="dt">template=</span><span class="st">"batchtools.slurm.tmpl"</span>, <span class="dt">runid=</span><span class="st">"01"</span>, <span class="dt">Njobs=</span><span class="dv">12</span>, <span class="dt">resourceList=</span>resources)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="kw">getStatus</span>(<span class="dt">reg=</span>hisat2)</a></code></pre></div>
<p>Check whether all BAM files have been created</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span>(args))</a></code></pre></div>
<p>Check all reads mapping results.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">read_statsDF_READS &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/alignStats.html">alignStats</a></span>(<span class="dt">args=</span>args)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(read_statsDF_READS, <span class="st">"results/alignStats_READS.xls"</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">sep=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a></code></pre></div>
</div>
</div>
<div id="mapping-unique-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-unique-reads" class="anchor"></a>Mapping Unique reads</h2>
<div id="collapsing-identical-sequences" class="section level3">
<h3 class="hasAnchor">
<a href="#collapsing-identical-sequences" class="anchor"></a>Collapsing identical sequences</h3>
<p>Collapsing identical sequences in a FASTQ/A file into a single sequence (while maintaining reads counts).</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/trim_collapser.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets_clip.txt"</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span>(args)[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">make_bam=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x=</span>args, <span class="dt">file=</span><span class="st">"targets_clip_collapser.txt"</span>, <span class="dt">overwrite=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<div id="reads-length-distribution-1" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution-1" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after collapsing the identical reads.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"seeDistribution.R"</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"results/Distribution_collapsing.pdf"</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">seeDistribution</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span>(args), <span class="dt">file=</span><span class="st">"fasta"</span>)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
</div>
<div id="removing-barcode" class="section level3">
<h3 class="hasAnchor">
<a href="#removing-barcode" class="anchor"></a>Removing Barcode</h3>
<p>Remove barcode after collapse.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/trim_fasta.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets_clip_collapser.txt"</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">make_bam=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x=</span>args, <span class="dt">file=</span><span class="st">"targets_clip_collap_trim.txt"</span>, <span class="dt">overwrite=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<div id="reads-length-distribution-2" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution-2" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after barcode removal.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"seeDistribution.R"</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"results/Distribution_Collapsing_barcodeRemoval.pdf"</span>)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw">seeDistribution</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span>(args), <span class="dt">file=</span><span class="st">"fasta"</span>)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
</div>
</div>
<div id="read-mapping-with-hisat2-1" class="section level3">
<h3 class="hasAnchor">
<a href="#read-mapping-with-hisat2-1" class="anchor"></a>Read mapping with <code>HISAT2</code>
</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma=</span><span class="st">"param/hisat2_unique.param"</span>, <span class="dt">mytargets=</span><span class="st">"targets_clip_collap_trim.txt"</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span>(args)[<span class="dv">1</span>] <span class="co"># Command-line parameters for first FASTQ file</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span>(args))</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5">resources &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">walltime=</span><span class="dv">1440</span>, <span class="dt">ntasks=</span><span class="dv">1</span>, <span class="dt">ncpus=</span><span class="dv">12</span>, <span class="dt">memory=</span><span class="dv">10240</span>, <span class="dt">email=</span><span class="st">"danielac@ucr.edu"</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">hisat2 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/clusterRun.html">clusterRun</a></span>(args, <span class="dt">conffile=</span><span class="st">".batchtools.conf.R"</span>, <span class="dt">template=</span><span class="st">"batchtools.slurm.tmpl"</span>, <span class="dt">runid=</span><span class="st">"01"</span>, <span class="dt">Njobs=</span><span class="dv">12</span>, <span class="dt">resourceList=</span>resources)</a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="kw">getStatus</span>(<span class="dt">reg=</span>hisat2)</a></code></pre></div>
<p>Check whether all BAM files have been created</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span>(args))</a></code></pre></div>
<p>Check all reads mapping results.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">read_statsDF_READS &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/alignStats.html">alignStats</a></span>(<span class="dt">args=</span>args)</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(read_statsDF_READS, <span class="st">"results/alignStats_READS_unique.xls"</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">sep=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a></code></pre></div>
</div>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R Under development (unstable) (2020-02-05 r77773)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 19.04
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so
## LAPACK: /usr/local/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] batchtools_0.9.12           ape_5.3                     ggplot2_3.2.1              
##  [4] systemPipeR_1.21.2          ShortRead_1.45.3            GenomicAlignments_1.23.1   
##  [7] SummarizedExperiment_1.17.2 DelayedArray_0.13.4         matrixStats_0.55.0         
## [10] Biobase_2.47.2              BiocParallel_1.21.2         Rsamtools_2.3.4            
## [13] Biostrings_2.55.4           XVector_0.27.0              GenomicRanges_1.39.2       
## [16] GenomeInfoDb_1.23.13        IRanges_2.21.3              S4Vectors_0.25.12          
## [19] BiocGenerics_0.33.0         BiocStyle_2.15.6           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1         rjson_0.2.20             hwriter_1.3.2           
##   [4] rprojroot_1.3-2          fs_1.3.1                 rstudioapi_0.11         
##   [7] bit64_0.9-7              AnnotationDbi_1.49.1     codetools_0.2-16        
##  [10] splines_4.0.0            knitr_1.28               annotate_1.65.1         
##  [13] GO.db_3.10.0             dbplyr_1.4.2             png_0.1-7               
##  [16] pheatmap_1.0.12          graph_1.65.1             BiocManager_1.30.10     
##  [19] compiler_4.0.0           httr_1.4.1               GOstats_2.53.0          
##  [22] backports_1.1.5          assertthat_0.2.1         Matrix_1.2-18           
##  [25] lazyeval_0.2.2           limma_3.43.4             htmltools_0.4.0         
##  [28] prettyunits_1.1.1        tools_4.0.0              gtable_0.3.0            
##  [31] glue_1.3.1               GenomeInfoDbData_1.2.2   Category_2.53.1         
##  [34] dplyr_0.8.4              rappdirs_0.3.1           Rcpp_1.0.3              
##  [37] pkgdown_1.4.1            vctrs_0.2.3              nlme_3.1-144            
##  [40] rtracklayer_1.47.0       xfun_0.12                stringr_1.4.0           
##  [43] lifecycle_0.1.0          XML_3.99-0.3             edgeR_3.29.0            
##  [46] zlibbioc_1.33.1          MASS_7.3-51.5            scales_1.1.0            
##  [49] BSgenome_1.55.3          VariantAnnotation_1.33.0 hms_0.5.3               
##  [52] RBGL_1.63.1              RColorBrewer_1.1-2       yaml_2.2.1              
##  [55] curl_4.3                 memoise_1.1.0            biomaRt_2.43.3          
##  [58] latticeExtra_0.6-29      stringi_1.4.6            RSQLite_2.2.0           
##  [61] genefilter_1.69.0        desc_1.2.0               checkmate_2.0.0         
##  [64] GenomicFeatures_1.39.5   rlang_0.4.4              pkgconfig_2.0.3         
##  [67] bitops_1.0-6             evaluate_0.14            lattice_0.20-38         
##  [70] purrr_0.3.3              bit_1.1-15.2             tidyselect_1.0.0        
##  [73] GSEABase_1.49.0          AnnotationForge_1.29.2   magrittr_1.5            
##  [76] bookdown_0.17            R6_2.4.1                 base64url_1.4           
##  [79] DBI_1.1.0                withr_2.1.2              pillar_1.4.3            
##  [82] survival_3.1-8           RCurl_1.98-1.1           tibble_2.1.3            
##  [85] crayon_1.3.4             BiocFileCache_1.11.4     rmarkdown_2.1           
##  [88] jpeg_0.1-8.1             progress_1.2.2           locfit_1.5-9.1          
##  [91] grid_4.0.0               data.table_1.12.8        blob_1.2.1              
##  [94] Rgraphviz_2.31.0         digest_0.6.24            xtable_1.8-4            
##  [97] brew_1.0-6               openssl_1.4.1            munsell_0.5.0           
## [100] askpass_1.1</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-H_Backman2016-bt">
<p>H Backman, Tyler W, and Thomas Girke. 2016. “systemPipeR: NGS workflow and report generation environment.” <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#background-and-objectives">Background and objectives</a></li>
      <li><a href="#experimental-design">Experimental design</a></li>
      </ul>
</li>
      <li>
<a href="#workflow-environment">Workflow environment</a><ul class="nav nav-pills nav-stacked">
<li><a href="#generate-workflow-environment">Generate workflow environment</a></li>
      <li><a href="#run-workflow">Run workflow</a></li>
      </ul>
</li>
      <li>
<a href="#read-preprocessing">Read preprocessing</a><ul class="nav nav-pills nav-stacked">
<li><a href="#experiment-definition-provided-by-targets-file">Experiment definition provided by <code>targets</code> file</a></li>
      <li><a href="#fastq-quality-report">FASTQ quality report</a></li>
      <li><a href="#read-quality-filtering-and-trimming">Read quality filtering and trimming</a></li>
      <li><a href="#mapping-all-reads">Mapping ALL reads</a></li>
      <li><a href="#mapping-unique-reads">Mapping Unique reads</a></li>
      </ul>
</li>
      <li><a href="#version-information">Version Information</a></li>
      <li><a href="#funding">Funding</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '',
    indexName: '',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
